atlasexec/copilot.go

```diff
@@ -0,0 +1,38 @@
+package atlasexec
+
+import (
+	"context"
+	"strings"
+)
+
+type (
+	CopilotParams struct{ Prompt, Session string }
+	// Copilot is the result of a Copilot execution.
+	Copilot []*CopilotMessage
+	// CopilotMessage is the JSON message emitted by the Copilot OneShot execution.
+	CopilotMessage struct {
+		// Session ID for the Copilot session.
+		SessionID string `json:"sessionID,omitempty"`
+		// Type of the message. Only "message" is currently supported.
+		Type string `json:"type"`
+		// Content of the message,
+		Content string `json:"content,omitempty"`
+	}
+)
+
+// Copilot executes a one-shot Copilot session with the provided options.
+func (c *Client) Copilot(ctx context.Context, params *CopilotParams) (Copilot, error) {
+	args := []string{"copilot", "-q", params.Prompt}
+	if params.Session != "" {
+		args = append(args, "-r", params.Session)
+	}
+	return jsonDecode[CopilotMessage](c.runCommand(ctx, args))
+}
+
+func (c Copilot) String() string {
+	var buf strings.Builder
+	for _, msg := range c {
+		buf.WriteString(msg.Content)
```

Comment:
Maybe we need check the message type

Reply:
![image](https://github.com/user-attachments/assets/0281c4c0-b092-44a7-8e2d-1620bd2b7bc8)

In a later stage :-)